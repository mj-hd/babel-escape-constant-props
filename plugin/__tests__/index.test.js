const babel = require('babel-core');
const plugin = require('../');

test.each([
  ['number', '<Foo prop={1} />'],
  ['boolean', '<Foo prop={true} />'],
  ['shorthanded boolean', '<Foo prop />'],
  ['string', '<Foo prop="bar" />'],
  ['null', '<Foo prop={null} />'],
  ['constant object contains string', "<Foo prop={{ foo: 'bar' }} />"],
  ['constant object contains number', '<Foo prop={{ foo: 1 }} />'],
  ['constant object contains null', '<Foo prop={{ foo: null }} />'],
  [
    'constant object contains constant object',
    "<Foo prop={{ foo: { foo: 'bar' } }} />",
  ],
  ['constant object contains array', '<Foo prop={{ foo: [1, 2] }} />'],
  [
    'constant object contains array of array',
    "<Foo prop={{ foo: [['foo', 'bar']] }} />",
  ],
  [
    'constant object contains array of constant objects',
    "<Foo prop={{ foo: [{ foo: 'bar' }, { hoge: 'piyo' }] }} />",
  ],
  ['constant array', '<Foo prop={[1, 2]} />'],
  [
    'constant array of constant objects',
    "<Foo prop={[{ foo: 'bar' }, { hoge: 'piyo' }]} />",
  ],
  ['identifier', '<Foo prop={hoge} />'],
  ["identifier as object's key", "<Foo prop={{ [hoge]: 'bar' }} />"],
  ['object contains identifier', '<Foo prop={{ foo: bar }} />'],
  ['function', '<Foo prop={func()} />'],
  ['arrow function', '<Foo prop={() => true} />'],
])('snapshot %s', (title, testcase) => {
  const {code} = babel.transform(testcase, {
    presets: ['@babel/preset-react'],
    plugins: [plugin],
  });
  expect(code).toMatchSnapshot();
});
