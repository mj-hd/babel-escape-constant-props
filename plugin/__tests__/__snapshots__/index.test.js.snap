// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot arrow function 1`] = `
"React.createElement(Foo, {
  prop: () => true
});"
`;

exports[`snapshot boolean 1`] = `
"React.createElement(Foo, {
  prop: true
});"
`;

exports[`snapshot constant array 1`] = `
"const _ref = [1, 2];
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant array of constant objects 1`] = `
"const _ref = [{
  foo: 'bar'
}, {
  hoge: 'piyo'
}];
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains array 1`] = `
"const _ref = {
  foo: [1, 2]
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains array of array 1`] = `
"const _ref = {
  foo: [['foo', 'bar']]
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains array of constant objects 1`] = `
"const _ref = {
  foo: [{
    foo: 'bar'
  }, {
    hoge: 'piyo'
  }]
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains constant object 1`] = `
"const _ref = {
  foo: {
    foo: 'bar'
  }
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains null 1`] = `
"const _ref = {
  foo: null
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains number 1`] = `
"const _ref = {
  foo: 1
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot constant object contains string 1`] = `
"const _ref = {
  foo: 'bar'
};
React.createElement(Foo, {
  prop: _ref
});"
`;

exports[`snapshot function 1`] = `
"React.createElement(Foo, {
  prop: func()
});"
`;

exports[`snapshot identifier 1`] = `
"React.createElement(Foo, {
  prop: hoge
});"
`;

exports[`snapshot identifier as object's key 1`] = `
"React.createElement(Foo, {
  prop: {
    [hoge]: 'bar'
  }
});"
`;

exports[`snapshot null 1`] = `
"React.createElement(Foo, {
  prop: null
});"
`;

exports[`snapshot number 1`] = `
"React.createElement(Foo, {
  prop: 1
});"
`;

exports[`snapshot object contains identifier 1`] = `
"React.createElement(Foo, {
  prop: {
    foo: bar
  }
});"
`;

exports[`snapshot shorthanded boolean 1`] = `
"React.createElement(Foo, {
  prop: true
});"
`;

exports[`snapshot string 1`] = `
"React.createElement(Foo, {
  prop: \\"bar\\"
});"
`;
